src/pool/srv_pool.c

```diff
@@ -1812,10 +1788,11 @@ ds_pool_connect_handler(crt_rpc_t *rpc)
 	if (rc != 0)
 		D_GOTO(out_map_version, rc);
 
-	if (!permitted(&ugm, in->pci_uid, in->pci_gid, in->pci_capas)) {
-		D_ERROR(DF_UUID": refusing connect attempt for uid %u gid %u "
-			DF_X64"\n", DP_UUID(in->pci_op.pi_uuid), in->pci_uid,
-			in->pci_gid, in->pci_capas);
+	rc = ds_check_pool_access(&ugm, &in->pci_cred, in->pci_capas);
+	if (rc != 0) {
+		D_ERROR(DF_UUID": refusing connect attempt for "
+			DF_X64" error: %d\n", DP_UUID(in->pci_op.pi_uuid),
+			in->pci_capas, rc);
```

Comment:
I assume that this is where we will fail if the agent isn't running, right?
If so, I would probably add a more meaningful error message since this mistake will probably be very frequent.
e.g. cannot connect to agent. is the agent running? or something similar.

Reply:
No you will get the failure in the client code not the server code.  It will usually come back as DER_BADPATH in the client code although there are other possible failure cases.